File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionYamlParse.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;
import co.nlighten.jsontransform.formats.yaml.YamlFormat;

public class TransformerFunctionYamlParse extends TransformerFunction {

    public TransformerFunctionYamlParse() {
        super();
    }

    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        return new YamlFormat(context.getAdapter()).deserialize(context.getString(null));
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionYamlParse.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import YamlFormat from "../formats/yaml/YamlFormat";

class TransformerFunctionYamlParse extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    if (str == null) return null;
    return YamlFormat.INSTANCE.deserialize(str);
  }
}

export default TransformerFunctionYamlParse;
